pyanaconda/storage/checker.py

```diff
@@ -411,6 +413,60 @@ def verify_mounted_partitions(storage, constraints, report_error, report_warning
                                "installation. Please unmount it and retry.") % part.path)
 
 
+def verify_lvm_destruction(storage, constraints, report_error, report_warning):
+    """Verify that destruction of LVM devices is correct.
+
+    This means: A VG and all its PVs must be all destroyed together, nothing is allowed to remain.
+
+    :param storage: a storage to check
+    :param constraints: a dictionary of constraints
+    :param report_error: a function for error reporting
+    :param report_warning: a function for warning reporting
+    """
+    # Implementation of the verification uses an inverted approach: Check that all PVs to be
+    # destroyed are contained in a VG that will be destroyed too. If there is a PV to be destroyed
+    # that has a VG that isn't going to be destroyed, that is an error. This is because we ask
+    # blivet to ignore some disks, and these might contain a PV. Crucially, blivet implements this
+    # such that the ignoring hides also all dependent devices - for any disk, it could be a PV
+    # partition, which would then hide also all VGs and LVs that depend on the PV. It does not
+    # matter that the VG needs other PVs too - the hiding wins.
+    destroyed_vg_names = set()
+    all_touched_vgs_and_disks = defaultdict(list)
+
+    for action in storage.devicetree.actions:
+        if action.is_destroy and action.is_device and action.device.type == "lvmvg":
+            destroyed_vg_names.add(action.device.name)
+
+        elif action.is_destroy and action.is_format and action.orig_format.type == "lvmpv":
+            # Check if the PV actually had any VG assigned
+            if action.orig_format.vg_name:
+                pv_name = action.device.name
+                disk_name = action.device.disk.name
+                vg_name = action.orig_format.vg_name
+                all_touched_vgs_and_disks[vg_name].append(disk_name)
```

Comment:
This implementation of the verification check is a little harder to follow. I liked the previous one.

Reply:
Mainly it's due to trying to find what *all* the involved disks are/were. I will remove that and make it more simple.